src/node/internal/streams_transform.js

```diff
@@ -172,7 +168,12 @@ Transform.prototype._write = function (chunk, encoding, callback) {
     if (val != null) {
       this.push(val);
     }
-    if (
+    if (rState.ended) {
+      // If user has called this.push(null) we have to
+      // delay the callback to properly propagate the new
+      // state.
+      nextTick(callback);
```

Comment:
this might also make this a breaking change...

Reply:
Altough @ronag considers this a semver-major change - https://github.com/nodejs/node/issues/53185#issuecomment-2136658496.

I'll put this behind a compat flag.